### PR TITLE
[ft-165412937] add reading stats

### PR DESCRIPTION
**Pull request by luc-tuyishime**
#### What does this PR do?
- Have CREATE and GET on resource endpoint in express REST API
#### Description of Task to be completed?
Have the following endpoints working
- CREATE api/v1/user/:articleSlug/stats
- PATCH api/v1/user/profile/stats

#### How should this be manually tested?
- It should be tested using POSTMAN
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- (#165412937)[https://www.pivotaltracker.com/story/show/165412937]
#### Screenshots (if appropriate)
- None
#### Questions:

### CODE REVIEW

File: src/controllers/ArticleController.js [hunk 1]
```diff
@@ -11,15 +11,16 @@ export default class ArticleController {
...
+    // const image = await helpers.upload(req);
     const { tagList } = req.body;
     const newArticle = await Article.create({
       userId: req.user.id || 0,
       slug: helpers.generator.slug(req.body.title),
       title: req.body.title.trim(),
       description: req.body.description.trim(),
       body: req.body.body.trim(),
-      coverUrl: image.image.original,
+      // coverUrl: image.image.original,
```
[the remark this answers is not in the document]
Comment (luc-tuyishime) [reply src/controllers/ArticleController.js]: Removed

File: src/controllers/ArticleController.js [hunk 2]
```diff
@@ -11,15 +11,16 @@ export default class ArticleController {
...
+      coverUrl: req.body.coverUrl || 'placeholder.png',
```
[the remark this answers is not in the document]
Comment (luc-tuyishime) [reply src/controllers/ArticleController.js]: Removed

File: src/controllers/UserController.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import dotenv from 'dotenv';
-import { User } from '../queries';
+import { User, findAndCountAll, create } from '../queries';
```
[the remark this answers is not in the document]
Comment (luc-tuyishime) [reply src/controllers/UserController.js]: findAndCountAll return both data and total count.

[the remark this answers is not in the document]
Comment (e-liyai) [reply src/controllers/UserController.js]: count for what? it should be named appropriately

[the remark this answers is not in the document]
Comment (luc-tuyishime) [reply src/controllers/UserController.js]: I have renamed to an appropriate name.

File: src/controllers/UserController.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import dotenv from 'dotenv';
-import { User } from '../queries';
+import { User, getAll, create } from '../queries';
```
Comment (e-liyai) [inline src/controllers/UserController.js]: this should be something like `getAllRatings`, use the object approach. Have an object ReadingStat and inside that have all the methods you want. That way you can name them the way you have it above. `getAll` and `Create`. This makes it easier to read and use. You will import the `ReadingStat` object and anywhere you want to use it you will invoke `ReadingStat.create` or `ReadingStat.getAll`

Comment (luc-tuyishime) [reply src/controllers/UserController.js]: ok

**Pull request merged**